.github/files/coverage-munger/post-message.sh

```diff
@@ -0,0 +1,163 @@
+COMMIT=$( jq --arg V "$PR_HEAD" -nr '$V | @uri' )
+
+# Fetch covinfo from server, if necessary.
+if [[ -z "$COVINFO" ]]; then
+	echo '::group::Fetching coverage info from server'
+	RES=$( curl -v \
+		--write-out '\nCode: %{response_code}' \
+		--url "https://jetpackcodecoverage.atomicsites.blog/get-pr-info.php?id=${ID}&commit=${COMMIT}"
+	)
+	STS=${RES##*$'\n'Code: }
+	if [[ "$STS" != 200 ]]; then
+		echo '::endgroup::'
+		echo "$RES"
+		[[ "$STS" == 404 || "$STS" == 409 ]]
+		exit $?
+	fi
+	COVINFO=${RES%$'\n'Code: *}
+	jq . <<<"$COVINFO"
+	echo '::endgroup::'
+else
+	echo '::group::Coverage info'
+	jq . <<<"$COVINFO"
+	echo '::endgroup::'
+fi
+
+# Find coverage run status, if necessary
+if [[ -z "$STATUS" ]]; then
+	echo "::group::Looking for latest coverage run"
+	PAGE=1
+	R=null
+	while true; do
+		J=$( curl -v -L fail \
+			--url "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/commits/${COMMIT}/check-runs?per_page=100&page=$PAGE" \
+			--header "authorization: Bearer $API_TOKEN_GITHUB"
+		)
+		R=$( jq --argjson R "$R" '[ ( $R, .check_runs[] ) | select( .name == "Code coverage" ) ] | sort_by( .completed_at // "running" ) | last' <<<"$J" )
+		if jq -e '.check_runs | length < 100' <<<"$J" &>/dev/null; then
+			break
+		fi
+		PAGE=$(( PAGE + 1 ))
+	done
+	jq . <<<"$R"
+	echo "::endgroup::"
+	STATUS=$( jq -r '.conclusion // .status // null' <<<"$R" )
+fi
+echo "Last run status is $STATUS"
+
+echo '::group::Checking labels for PR'
+LABELS=$( curl -v -L --fail \
+	--url "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/issues/${ID}/labels?per_page=100" \
+	--header "authorization: Bearer $API_TOKEN_GITHUB"
+)
+jq . <<<"$LABELS"
+echo "::endgroup::"
+
+echo "::group::Setting GitHub status"
+if jq -e '.covinfo' <<<"$COVINFO" &>/dev/null; then
+	COVINFO=$( jq '.covinfo' <<<"$COVINFO" )
+	if [[ "$STATUS" != 'success' ]]; then
+		COVINFO=$( jq '.state |= "pending" | .description |= "Waiting for tests to pass" | .msg |= "Cannot generate coverage summary while tests are failing. :zipper_mouth_face:\n\nPlease fix the tests, or re-run the Code coverage job if it was something being flaky."' <<<"$COVINFO" )
+	fi
+else
+	COVINFO='{"state":"error","description":"No covinfo received from server","msg":"","footer":""}'
+fi
+# If we're here from a label or unlabel event, and the CI check is still running, then no message.
+if [[ "$STATUS" == 'in_progress' ]]; then
+	COVINFO=$( jq '.state |= "pending" | .description |= "Waiting for tests to pass" | .msg |= ""' <<<"$COVINFO" )
+fi
+# If the "I don't care" label is set, override the status.
+if jq -e --arg LABEL "$SKIPLABEL" '.[] | select( .name == $LABEL )' <<<"$LABELS" &>/dev/null; then
+	COVINFO=$( jq --arg REPO "$GITHUB_REPOSITORY" --arg LABEL "$SKIPLABEL" '.state |= "success" | .description |= "Overridden by label" | .footer += "\n\n<sub>Coverage check overridden by https://github.com/\( $REPO | @uri | gsub("%2F"; "/") )/labels/\( $LABEL | @uri ).</sub>"' <<<"$COVINFO" )
+fi
```

Comment:
Would it make sense to return early if it's skip-labelled? If we don't care about coverage, we also don't care about all the other stuff, right?

Reply:
We still need to set the status check (lines 98–107). And personally I think still posting the message would be useful even if it's not blocking the merge.